incubator/hnc/docs/releasing.md

```diff
@@ -0,0 +1,140 @@
+the release branch._ The user guide should contain instructions for at least the
+last two minor releases of HNC - e.g., if you're about to release v0.6, do not
+remove documentation for HNC v0.4 yet.
+
+## Create a release branch
+
+Assuming that the Github repo is your `upstream` remote, simply do the following
+steps:
+
+```bash
+# Make sure you don't have any modified files in your repo.
+$ git checkout master
+$ git pull
+$ git checkout -b <branch name>
+$ git push upstream <branch name>
+```
+
+Do _not_ create a pull request for this branch! Instead, if you're a repo
+administrator, go to the Github UI and mark the branch as protected, which will
+prevent people from deleting it by accident. Otherwise, ask a repo admin to do
+this for you.
+
+## Create a release
+
+### Set up your environment
+
+Set the following environment variables:
+
+```bash
+export MT_ENDPOINT=https://api.github.com/repos/kubernetes-sigs/multi-tenancy
+export HNC_USER=<your github name>
+export HNC_PAT=<your personal access token>
+export HNC_IMG_TAG=<the semantic version, eg v0.1.0-rc1>
+```
+
+Note that `HNC_IMG_TAG` does _not_ include the `hnc-` prefix. That is because
+the image tag will only apply to HNC images, while the _Git_ tag (and branch)
+names apply to this repo, which includes non-HNC projects.
+
+### Create a release in Github
+
+1. Ensure that the Github tag name is `hnc-$HNC_IMG_TAG`, like `hnc-v0.1.0-rc1`.
+2. Follow the pattern in earlier releases in the description - e.g. include
+   installation instructions, key new features, a detailed change log, and known
+   issues.
+3. Add a "**RELEASE IN PROGRESS, DO NOT USE**" warning to the top of the
+   description.
+4. If this is a release candidate, mark the release as pre-production.
+5. Save the release. This will create a tag in the git repo.
+
+### Get the release ID
+
+Get the release ID. Sadly, this isn't available through the UI, so get it by
+calling:
+
+```bash
+curl -u "$HNC_USER:$HNC_PAT" $MT_ENDPOINT/releases | less`
+```
+
+Finding your release, and noting it's ID. Save this as an env var:
+
+`export HNC_RELEASE_ID=<id>`
+
+### Build the image and manifests
+
+From your local repo, call `make release`. This will use `gcloud` to submit a
+job to Cloud Build. Note that your local files are _not_ used to build HNC;
+Cloud Build will pull them directly from Github using your personal access
+token.
+
+**Note that your personal access token will be visible in the build logs,** but
+will not be printed to the console from `make` itself. TODO: [fix
+this](https://cloud.google.com/cloud-build/docs/securing-builds/use-encrypted-secrets-credentials#example_build_request_using_an_encrypted_variable).
+
+## Test
+
+Test! At a minimum, install it onto a cluster, download the kubectl plugin, and
+ensure that you can create a subnamespace. Better yet, run through the demo and
+all the `hack/test-*.sh` scripts. _TODO: automate this_
```

Comment:
Maybe add a link to the e2e test issue #785 ?

Reply:
Done